File: 240514/정수 명령 처리 3/process-numeric-commands-3.java
import java.util.*;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int n = sc.nextInt();
    static ArrayDeque<Integer> deque = new ArrayDeque<Integer>();
    public static void main(String[] args) {
        for(int i = 0; i < n; i++) {
            String cmd = sc.next();
            switch(cmd) {
                case "push_front":
                    deque.addFirst(sc.nextInt());
                    break;
                case "push_back":
                    deque.addLast(sc.nextInt());
                    break;
                case "pop_front":
                    System.out.println(deque.pollFirst());
                    break;
                case "pop_back":
                    System.out.println(deque.pollLast());
                    break;
                case "size":
                    System.out.println(deque.size());
                    break;
                case "empty":
                    if(deque.isEmpty()) {
                        System.out.println(1);
                    }else {
                        System.out.println(0);
                    }
                    break;
                case "front":
                    System.out.println(deque.peekFirst());
                    break;
                case "back":
                    System.out.println(deque.peekLast());
                    break; 
            }
        }
    }
}

File: 240514/정수 명령 처리 3/process-numeric-commands-3.kt
import java.util.*

fun main() {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val deque = ArrayDeque<Int>()

    for(i in 1.. n) {
        val cmd = sc.next()
        when(cmd) {
            "push_front" -> { 
                deque.addFirst(sc.nextInt())
             }
            "push_back" -> { 
                deque.addLast(sc.nextInt())
             }
            "pop_front" -> { 
                println(deque.pollFirst())
             }
            "pop_back" -> { 
                println(deque.pollLast())
             }
            "size" -> { 
                println(deque.size)
             }
            "empty" -> { 
                if(deque.isEmpty()) {
                    println(1)
                }else{
                    println(0)
                }
             }
            "front" -> { 
                println(deque.peekFirst())
             }
            "back" -> { 
                println(deque.peekLast())
             }
        }
    }
}